pulsar-io/netty/src/main/java/org/apache/pulsar/io/netty/server/NettyServer.java

```diff
@@ -108,6 +116,22 @@ private void runTcp() throws InterruptedException {
         channelFuture.channel().closeFuture().sync();
     }
 
+    private void runHttp() throws InterruptedException {
+        bossGroup = new NioEventLoopGroup(this.numberOfThreads);
+        workerGroup = new NioEventLoopGroup(this.numberOfThreads);
+
+        ServerBootstrap serverBootstrap = new ServerBootstrap();
+        serverBootstrap.group(bossGroup, workerGroup)
+                .channel(NioServerSocketChannel.class)
+                .childHandler(new NettyHttpChannelInitializer(new NettyHttpServerHandler(this.nettySource), null))
+                .option(ChannelOption.SO_BACKLOG, 1024)
+                .childOption(ChannelOption.SO_KEEPALIVE, true);
+
+        ChannelFuture channelFuture = serverBootstrap.bind(this.host, this.port).sync();
+        channelFuture.channel().closeFuture().sync();
+    }
+
```

Comment:
we can define getServerBootstrap to be reused by runTcp() and runHttp(), like :
```java
private ServerBootstrap getServerBootstrap(ChannelHandler childHandler) {
        bossGroup = new NioEventLoopGroup(this.numberOfThreads);
        workerGroup = new NioEventLoopGroup(this.numberOfThreads);
        ServerBootstrap serverBootstrap = new ServerBootstrap();
        serverBootstrap.group(bossGroup, workerGroup);
        serverBootstrap.channel(NioServerSocketChannel.class);
        serverBootstrap.childHandler(childHandler)
                .option(ChannelOption.SO_BACKLOG, 1024)
                .childOption(ChannelOption.SO_KEEPALIVE, true);

        return serverBootstrap;
    }
```
How about that?

Reply:
Yes, I wasn't a fan of having duplicate code, so I will rework that piece a bit.

Reply:
Actually the runUDP method uses the Bootstrap class while the other two use the ServerBootstrap class.  I am not sure why that is the case, but it does make the above suggestion a bit more tricky